homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -1,102 +1,97 @@
-DEFAULT_AVOID_TOLL_ROADS = False
-DEFAULT_AVOID_SUBSCRIPTION_ROADS = False
-DEFAULT_AVOID_FERRIES = False
+SCAN_INTERVAL = timedelta(minutes=5)
 
-ICON = "mdi:car"
 
-UNITS = [CONF_UNIT_SYSTEM_METRIC, CONF_UNIT_SYSTEM_IMPERIAL]
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(WAZE_SCHEMA)
 
-REGIONS = ["US", "NA", "EU", "IL", "AU"]
-VEHICLE_TYPES = ["car", "taxi", "motorcycle"]
 
-SCAN_INTERVAL = timedelta(minutes=5)
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Waze travel time sensor platform."""
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_ORIGIN): cv.string,
-        vol.Required(CONF_DESTINATION): cv.string,
-        vol.Required(CONF_REGION): vol.In(REGIONS),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_INCL_FILTER): cv.string,
-        vol.Optional(CONF_EXCL_FILTER): cv.string,
-        vol.Optional(CONF_REALTIME, default=DEFAULT_REALTIME): cv.boolean,
-        vol.Optional(CONF_VEHICLE_TYPE, default=DEFAULT_VEHICLE_TYPE): vol.In(
-            VEHICLE_TYPES
+    asyncio.run_coroutine_threadsafe(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
         ),
-        vol.Optional(CONF_UNITS): vol.In(UNITS),
-        vol.Optional(
-            CONF_AVOID_TOLL_ROADS, default=DEFAULT_AVOID_TOLL_ROADS
-        ): cv.boolean,
-        vol.Optional(
-            CONF_AVOID_SUBSCRIPTION_ROADS, default=DEFAULT_AVOID_SUBSCRIPTION_ROADS
-        ): cv.boolean,
-        vol.Optional(CONF_AVOID_FERRIES, default=DEFAULT_AVOID_FERRIES): cv.boolean,
-    }
-)
+        hass.loop,
+    )
 
+    _LOGGER.info(
+        "Your Waze configuration has been imported into the UI; "
+        "please remove it from configuration.yaml as support for it "
+        "will be removed in a future release."
+    )
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Waze travel time sensor platform."""
-    destination = config.get(CONF_DESTINATION)
-    name = config.get(CONF_NAME)
-    origin = config.get(CONF_ORIGIN)
-    region = config.get(CONF_REGION)
-    incl_filter = config.get(CONF_INCL_FILTER)
-    excl_filter = config.get(CONF_EXCL_FILTER)
-    realtime = config.get(CONF_REALTIME)
-    vehicle_type = config.get(CONF_VEHICLE_TYPE)
-    avoid_toll_roads = config.get(CONF_AVOID_TOLL_ROADS)
-    avoid_subscription_roads = config.get(CONF_AVOID_SUBSCRIPTION_ROADS)
-    avoid_ferries = config.get(CONF_AVOID_FERRIES)
-    units = config.get(CONF_UNITS, hass.config.units.name)
+    return True
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up a Waze travel time sensor entry."""
+    destination = config_entry.data[CONF_DESTINATION]
+    name = config_entry.data.get(CONF_NAME)
```

Comment:
Can we use the origin and destination to generate a default name when one isn't provided?